accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,265 @@
+- Feature Name: Prometheus TLS support
+- Start Date: 2021-02-11
+- RFC PR:
+
+# Summary
+
+[summary]: #summary
+
+Add optional support for TLS encrypted and authenticated connection to
+Prometheus server and exporters.
+
+# Motivation
+
+[motivation]: #motivation
+
+## Prometheus
+
+In the default configuration it is presumed that untrusted users have access to
+the Prometheus HTTP endpoint. They have access to all time series information
+contained in the database, plus a variety of operational/debugging
+information [[1]](#1).
+
+Applications needing access to Prometheus API:
+
+* _Grafana_
+
+Users needing access to Prometheus API/UI:
+
+* monitoring administrator
+
+## Exporters
+
+Exporters generally only talk to one configured client instance with a preset
+set of commands/requests, which cannot be expanded via their HTTP endpoint.
+Still, all exposed metrics are available to untrusted users.
+
+> Uyuni should optionally support and automate setting up secure
+> configuration of Prometheus and exporters.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Prometheus
+
+Prometheus 2.24.0 has introduced TLS encryption as well as basic and
+client-certificate authentication [[2]](#2). All available configuration options are
+described in official documentation [[3]](#3). Basic authentication credentials are
+stored in the configuration file. Passwords are hashed with _bcrypt_.
+
+To follow current SUSE Manager security model [[4]](#4) we should consider
+implementing following default configuration:
+
+* Upgrade Prometheus to version >= 2.24.1.
+* [Optional] Create intermediate certificate authority (CA) dedicated for monitoring.
+* [Optional] Create and deploy Prometheus server certificate signed by this CA.
+* Configure `tls_server_config` using this certificate.
+* Configure basic authentication accounts to map SUSE Manager credentials.
```

Comment:
Does this mean that a static configuration file with all credential pairs would be generated at formula application time, based on accounts known from the DB, and regenerated at highstate time?

Reply:
Thanks for catching this. I think that would be too complicated and error prone. I've changed that point and proposed setting up a dedicated administrator account.